resolwe_bio/processes/workflows/quantseq.py

```diff
@@ -0,0 +1,369 @@
+                    description="Use the fraction of reads [0 - 1.0] from the orignal input file instead "
+                    "of the absolute number of reads. If set, this will override the"
+                    "'Number of reads' input parameter.",
+                )
+                two_pass = BooleanField(
+                    label="2-pass mode",
+                    default=False,
+                    description="Enable two-pass mode when down-sampling. Two-pass mode is twice "
+                    "as slow but with much reduced memory.",
+                )
+
+            advanced = GroupField(Advanced, label="Advanced options for downsampling")
+
+        cutadapt = GroupField(
+            Cutadapt, label="Cutadapt filtering", hidden="trimming_tool != 'cutadapt'"
+        )
+        downsampling = GroupField(
+            Downsampling,
+            label="Downsampling with Seqtk",
+        )
+        preprocessing = GroupField(
+            Preprocessing,
+            label="Preprocessing with BBDuk",
+            hidden="trimming_tool != 'bbduk'",
+        )
+
+    class Output:
+        """Output fields."""
+
+    def run(self, inputs, outputs):
+        """Run the workflow."""
+
+        if inputs.trimming_tool == "bbduk" and not inputs.adapters:
+            self.error(
+                "Please provide fasta file of adapters, if you want to use BBDuk as a trimming tool."
+            )
+
+        if inputs.adapters:
+
+            input_preprocessing = {
+                "reads": inputs.reads,
+                "min_length": 20,
+                "reference": {"sequences": inputs.adapters},
+                "processing": {"kmer_length": 13},
+                "operations": {
+                    "k_trim": "r",
+                    "min_k": 6,
+                    "quality_trim": "r",
+                    "trim_quality": 10,
+                    "quality_encoding_offset": inputs.preprocessing.quality_encoding_offset,
+                    "ignore_bad_quality": inputs.preprocessing.ignore_bad_quality,
+                },
+                "fastqc": {
+                    "nogroup": True,
+                },
+            }
+
+            if inputs.reads.type.startswith("data:reads:fastq:single:"):
+                process_slug = "bbduk-single"
+            elif inputs.reads.type.startswith("data:reads:fastq:paired:"):
+                process_slug = "bbduk-paired"
+            else:
+                self.error("Wrong reads input type was provided.")
+
+        else:
+            if inputs.reads.type.startswith("data:reads:fastq:single:"):
+                input_preprocessing = {
+                    "reads": inputs.reads,
+                    "options": {
+                        "quality_cutoff": inputs.cutadapt.quality_cutoff,
+                    },
+                }
+                process_slug = "cutadapt-3prime-single"
+            else:
+                self.error("Wrong reads input type was provided.")
```

Comment:
`self.error("Only single-end reads are supported  when Cutadapt is selected as a trimming tool.")`